grimblast/grimblast

```diff
@@ -148,14 +148,21 @@ elif [ "$SUBJECT" = "output" ] ; then
   OUTPUT=$(hyprctl monitors -j | jq -r '.[] | select(.active == "yes")' | jq -r '.name')
   WHAT="$OUTPUT"
 elif [ "$SUBJECT" = "window" ] ; then
+  SIZE=$(hyprctl monitors -j | jq -r 'map(.width,.height)')
   WORKSPACES="$(hyprctl monitors -j | jq -r 'map(.activeWorkspace.id)')"
-  WINDOWS="$(hyprctl clients -j | jq -r --argjson workspaces "$WORKSPACES" 'map(select([.workspace.id] | inside($workspaces)))' )"
-  GEOM=$(echo "$WINDOWS" | jq -r '.[] | "\(.at[0]),\(.at[1]) \(.size[0])x\(.size[1])"' | slurp)
-  # Check if user exited slurp without selecting the area
-  if [ -z "$GEOM" ]; then
-   exit 1
+  # Check if user is in fullscreen
+  if [ -z "$(hyprctl clients -j | jq -r ".[] |select (.workspace.id == "$(echo $WORKSPACES | jq '.[]')" and .size == $SIZE)")" ] ; then
+    WINDOWS="$(hyprctl clients -j | jq -r --argjson workspaces "$WORKSPACES" 'map(select([.workspace.id] | inside($workspaces)))' )"
```

Comment:
These two calls are redundant (and slow). We should instead set a variable
```sh
CLIENTS=$(hyprctl clients -j)
```
and pass it to `jq` with `<<< $CLIENTS`